src/mprk.jl

```diff
@@ -861,7 +862,9 @@ function perform_step!(integrator, cache::MPRK43ConstantCache, repeat_step = fal
 
     # avoid division by zero due to zero Patankar weights
     σ = add_small_constant(uprev, small_constant)
-    σ0 = σ
+    if !(q1 ≈ q2)
+        σ0 = copy(σ)
+    end
```

Comment:
Why do you need the copy here?

Reply:
Doesn't `σ0 = σ` create a reference to `σ` and hence `σ0` gets changed when `σ` is?

Reply:
We're just changing it via `=` assignments below - this does not create the problems you thought about here:

```julia
julia> let
       σ = 1
       σ0 = σ
       σ = 2
       σ, σ0
       end
(2, 1)

julia> let
       σ = SVector(1, 2)
       σ0 = σ
       σ = SVector(2, 3)
       σ, σ0
       end
([2, 3], [1, 2])

julia> let
       σ = [1, 2]
       σ0 = σ
       σ = [2, 3]
       σ, σ0
       end
([2, 3], [1, 2])
```

Reply:
It would only be problematic if you changed something in-place, e.g.,

```julia
julia> let
       σ = [1, 2]
       σ0 = σ
       σ .= [2, 3]
       σ, σ0
       end
([2, 3], [2, 3])

julia> let
       σ = [1, 2]
       σ0 = σ
       σ[1] = 5
       σ, σ0
       end
([5, 2], [5, 2])
```

But this doesn't happen in this out-of-place version.